addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/PlugwiseStickDiscoveryService.java

```diff
@@ -0,0 +1,272 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.plugwise.internal;
+
+import static org.openhab.binding.plugwise.PlugwiseBindingConstants.THING_TYPE_STICK;
+import static org.openhab.binding.plugwise.internal.protocol.field.DeviceType.STICK;
+
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.plugwise.PlugwiseBindingConstants;
+import org.openhab.binding.plugwise.internal.PlugwiseCommunicationHandler.MessagePriority;
+import org.openhab.binding.plugwise.internal.config.PlugwiseStickConfig;
+import org.openhab.binding.plugwise.internal.listener.PlugwiseMessageListener;
+import org.openhab.binding.plugwise.internal.protocol.InformationRequestMessage;
+import org.openhab.binding.plugwise.internal.protocol.InformationResponseMessage;
+import org.openhab.binding.plugwise.internal.protocol.Message;
+import org.openhab.binding.plugwise.internal.protocol.NetworkStatusRequestMessage;
+import org.openhab.binding.plugwise.internal.protocol.NetworkStatusResponseMessage;
+import org.openhab.binding.plugwise.internal.protocol.field.MACAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+import gnu.io.CommPortIdentifier;
+
+/**
+ * Discovers Stick devices by periodically sending a {@link NetworkStatusRequestMessage} to unused serial ports.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+public class PlugwiseStickDiscoveryService extends AbstractDiscoveryService
+        implements ExtendedDiscoveryService, PlugwiseMessageListener {
+
+    private static final Set<ThingTypeUID> DISCOVERED_THING_TYPES_UIDS = Sets.newHashSet(THING_TYPE_STICK);
+
+    private static final int DISCOVERY_INTERVAL = 180;
+    private static final int SCAN_TIMEOUT = 5;
+
+    private final Logger logger = LoggerFactory.getLogger(PlugwiseStickDiscoveryService.class);
+
+    private DiscoveryServiceCallback discoveryServiceCallback;
+
+    private ScheduledFuture<?> discoveryJob;
+
+    private PlugwiseCommunicationHandler communicationHandler = new PlugwiseCommunicationHandler();
+    private PlugwiseStickConfig stickConfig = new PlugwiseStickConfig();
```

Comment:
It feels strange to have this as a field, can it just be a local variable? Or maybe just move the initialization to a sensible place.